include/extractor/guidance/is_through_street.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef OSRM_EXTRACTOR_GUIDANCE_IS_THROUGH_STREET_HPP_
+#define OSRM_EXTRACTOR_GUIDANCE_IS_THROUGH_STREET_HPP_
+
+#include "extractor/guidance/constants.hpp"
+#include "extractor/guidance/sliproad_handler.hpp"
+#include "util/assert.hpp"
+#include "util/bearing.hpp"
+#include "util/coordinate_calculation.hpp"
+#include "util/guidance/name_announcements.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <iterator>
+#include <limits>
+
+#include <boost/assert.hpp>
+
+using osrm::extractor::guidance::getTurnDirection;
+using osrm::util::angularDeviation;
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+template <typename IntersectionType>
+inline bool isThroughStreet(const std::size_t index,
+                            const IntersectionType &intersection,
+                            const util::NodeBasedDynamicGraph &node_based_graph,
+                            const EdgeBasedNodeDataContainer &node_data_container,
+                            const util::NameTable &name_table,
+                            const SuffixTable &street_name_suffix_table)
+{
+    if (index >= intersection.size())
+    {
+        return false;
+    }
+
+    const auto &data_at_index = node_data_container.GetAnnotation(
+        node_based_graph.GetEdgeData(intersection[index].eid).annotation_data);
+
+    if (data_at_index.name_id == EMPTY_NAMEID)
+        return false;
+
+    // a through street cannot start at our own position -> index 1
+    for (std::size_t road_index = 1; road_index < intersection.size(); ++road_index)
+    {
+        if (road_index == index)
+            continue;
+
+        const auto &road = intersection[road_index];
+        const auto &road_data = node_data_container.GetAnnotation(
+            node_based_graph.GetEdgeData(road.eid).annotation_data);
+
+        // roads have a near straight angle (180 degree)
+        const bool is_nearly_straight = angularDeviation(road.angle, intersection[index].angle) >
+                                        (STRAIGHT_ANGLE - FUZZY_ANGLE_DIFFERENCE);
+
+        const bool have_same_name =
+            road_data.name_id != EMPTY_NAMEID &&
+            !util::guidance::requiresNameAnnounced(
+                data_at_index.name_id, road_data.name_id, name_table, street_name_suffix_table);
```

Comment:
Here and above: I remember @karenzshea changing how we compare name ids - could you give this a look?

Reply:
@daniel-j-h is this the change you mean that @karenzshea made? https://github.com/Project-OSRM/osrm-backend/pull/2963

Reply:
Looking at #2963 it doesn't seem like there's any change that I need to make to my name check above. However, I found this function in the rest of the codebase, and maybe that is what you are referring to? https://github.com/Project-OSRM/osrm-backend/blob/6d801e7086cf7c12384611bda4f441ac3ce4dc17/include/extractor/guidance/mergable_road_detector.hpp#L84

Reply:
Hmm okay so `MergableRoadDetector::HaveIdenticalNames` does the same exact thing and it would be great to use it, but it's a private function inside `MergeableRoadDetector`. Grrr.
Should I move it out and make it a function usable by all parties in Guidance?

Also, sidebar, but the misspelling of `mergeable` in the class name above is starting to bother me 😄